doc/python/b64.md

```diff
@@ -0,0 +1,132 @@
+---
+jupyter:
+  jupytext:
+    notebook_metadata_filter: all
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.3
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+  language_info:
+    codemirror_mode:
+      name: ipython
+      version: 3
+    file_extension: .py
+    mimetype: text/x-python
+    name: python
+    nbconvert_exporter: python
+    pygments_lexer: ipython3
+    version: 3.10.0
+  plotly:
+    description: How to use typed arrays in Plotly.py
+    display_as: file_settings
+    language: python
+    layout: base
+    name: Improving Performance with Typed Arrays
+    order: 39
+    permalink: python/b64/
+    thumbnail: thumbnail/b64.png
+---
+
+*New in Plotly.py 6.0**
+
+Improve the performance of generating Plotly figures that use a large number of data points by using NumPy arrays and other objects that can be converted to NumPy arrays, such as Pandas Series and Index objects. 
+
+Plotly.py uses Plotly.js for rendering, which supports base64-encoded typed arrays. In Plotly.py, NumPy array and NumPy-convertible arrays are base64 encoded before being passed to Plotly.js for rendering.
+
+
+## Arrays and Data Types Supported
+
+The following types of array objects in Python are supported:
+
+- Numpy `numpy.ndarray` objects.
+- Pandas Index, `pandas.Index`, or Series, `pandas.Series`, objects.
+- Array objects that can be converted to `numpy.ndarray` objects. i.e., they implement `"__array__"` or `"__array_interface__"` and return a `numpy.ndarray.
+
+The following array data types are supported:
+
+- int8
+- uint8
+- int16
+- uint16
+- int32
+- uint32
+- float32
+- float64
+- int64*
+- uint64*
+    
+*If the array dtype is **int64** and **uint64**, often the default dtype for arrays in NumPy when no dtype is specified, those dtypes will be changed to other types internally by Plotly.py where possible. 
+
+
+
+## Unsupported Attributes
+
+Arrays passsed to attributes with the following names do not use the Plotly.js base64 typed arrays functionality:
```

Comment:
I wonder if we should even bring up base64 here. Maybe it could just say something like "performance improvements related to using pandas and numpy arrays"?

Reply:
Along the same lines, maybe we could rename this file

Reply:
You're right. That info probably isn't that useful for the user. And I'll rename the file too. Thanks @marthacryan

Reply:
I made those updates. There's still a mention here. Is it base64-encoded typed arrays support in Plotly.js specifically that allows for the improved performance?

```
Plotly.py uses Plotly.js for rendering, which supports base64-encoded typed arrays. In Plotly.py, NumPy array and NumPy-convertible arrays are base64 encoded before being passed to Plotly.js for rendering.
```